### DISCUSSION TITLE
Examples using tanstack router + query with auth

### DISCUSSION THREAD

**Discussion by wolflu05** [TanStack/router#1092] [Q&A]
Hello, I already tried out tan stack router a long time ago, but now I came back to it and it now supports file based routing really cool. I take a look at the kitchen sink example also using tanstack query, but I'm not really getting how I would implement auth and their state management. The kitchen sink example just uses an object which is not the react way. I hope someone can point me into the right direction showing how to use e.g. react context, sod, ...for storing the API token, ... .

And secondly I'm not getting how I would implement a router where basically every page except the login page is behind the authenticated area.

I would really appreciate it if somehone could show/link me some examples showing the described things.
Lukas

**Comment by aaronlin0122** [comment]
still looking for best practice, here is my examples:
https://github.com/nekochan0122/tanstack-authentication
